src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/DampedHarmonic.cpp

```diff
@@ -257,39 +279,66 @@ void damped_harmonic_impl(
     get<0>(d4_RW_L2) += get(weight) * d0_roll_on;
   }
 
+  time_deriv_of_lapse<DataVector, SpatialDim, Frame>(
+      make_not_null(&dt_lapse), lapse, shift, spacetime_unit_normal, phi, pi);
+  spatial_deriv_of_lapse<DataVector, SpatialDim, Frame>(
+      make_not_null(&d3_lapse), lapse, spacetime_unit_normal, phi);
```

Comment:
You can reduce the work here by doing `-lapse * half_phi_two_normals`

Reply:
I elided d3_lapse all together.